File: src/SynthShop.UI/src/app/models/order.ts

export interface OrderItem {
    orderItemID: string;
    productID: string;
    quantity: number;
    price: number;
}

export interface Order {
    orderID: string;
    orderDate: Date;
    userId: string;
    isDeleted: boolean;
    totalAmount: number;
    status: number;
    createdAt: Date;
    updateAt?: Date;
    orderItems: OrderItem[];
}

File: src/SynthShop.Domain/Entities/Order.cs
using SynthShop.Domain.Enums;

namespace SynthShop.Domain.Entities;

public class Order
{
    public Guid OrderID { get; set; }
    public DateTime OrderDate { get; set; }
    public Guid UserId { get; set; }
    public bool IsDeleted { get; set; }
    public decimal TotalAmount { get; set; }
    public OrderStatus Status { get; set; }
    public User User { get; set; }
    public DateTime CreatedAt { get; set; }
    public DateTime? UpdateAt { get; set; }
    public ICollection<OrderItem> OrderItems { get; set; }
}